src/main/java/com/uid2/shared/store/reader/RotatingClientKeyProvider.java

```diff
@@ -55,14 +58,20 @@ public long getVersion(JsonObject metadata) {
 
     @Override
     public long loadContent(JsonObject metadata) throws Exception {
-        return reader.loadContent(metadata, "client_keys");
+        long version = reader.loadContent(metadata, "client_keys");
+        this.authorizableStore.refresh(this.getAll());
+        return version;
     }
 
     @Override
     public ClientKey getClientKey(String token) {
         return reader.getSnapshot().get(token);
     }
 
+    public ClientKey getClientKeyFromHash(String hash) {
```

Comment:
Should we update the `IClientKeyProvider` interface with the above method?

Reply:
Yep, done